src/wazuh_modules/wm_database.c

```diff
@@ -401,6 +407,20 @@ void wm_sync_agents() {
 
 #endif // LOCAL
 
+void wm_get_os_arch(char * os_header, char *os_arch) {
+    if      (strstr(os_header, "x86_64") != NULL) snprintf(os_arch, strlen("x86_64")+1, "%s", "x86_64");
```

Comment:
This function does not know the actual size of the `os_header` variable. `strlen("x86_64")` does not help since it's not related to the length of the destination string.

If you want to copy the first N bytes from a string, I suggest:
```c
void wm_get_os_arch(char * os_header, size_t size, char *os_arch) {
    if (strstr(os_header, "x86_64") != NULL) 
        snprintf(os_arch, size, "%.*s", strlen("x86_64"), "x86_64");
```

Although, if I understand well, you want to find a substring a return it. Why don't you use...
```c
char * wm_get_os_arch(char * os_header) {
    if (strstr(os_header, "x86_64")) return strdup("x86_64");
    // ...etc...
```